### DISCUSSION TITLE
RegExp: Insert space at the beginning of line matched with lookarounds

### DISCUSSION THREAD

**Discussion by PiotrMP006** [community/community#60132] [Programming Help]
### Body

How to insert a space at the beginning of a line in a "for next" code loop (.NET regular expressions)

I am trying to add spaces at the beginning of lines matched with look arounds

```
for
line of code
line of code
line of code
line of code
next
```

and this is the output I want to get

```
for
  line of code
  line of code
  line of code
  line of code
next
```

Please help me with .NET regular expressions

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by CodeMoreira** [comment]
### Explain the regex
The following regex get all lines between **for** and **next** `for\n((.+\n)+)next`, but it just get the lines and nothing more, if you want to add tabs before the line you'll need to create a class to do that.

### Create the class
The following code will do what you want.
```cs
// the full string
var text = @"for
line of code
line of code
line of code
line of code
next";

// regex
var pattern = @"for\n((.+\n)+)next";

// apply the regex to the string and don't forget to "import" System.Text.RegularExpressions
var replaced = Regex.Replace(text, pattern, "$1");

// when we get the lines the regex also get an empty space at the end, to remove this empty space just write the following code
splitContent = splitContent.Take(splitContent.Count() - 1).ToArray();

// now we will put all together

// create an string with "for" text that we removed
string finalText = "for";

// loop through the array of lines we got
foreach (var line in splitContent)
{
  // concat each line with finaltext  and add the tab
  
  // for add tab just write "\t" and to go to the next line write "\n"
  finalText += $"\n\t{line}";
};
// now add the text "next" that we removed
finalText += "\nnext";
```

### Example
I did this simple example to test the code https://onecompiler.com/csharp/3zdn5rewg

### Conclusion
Sorry if there was any mistake, I don't program in C#, I think this code is a great starting point to help you solve your problem and any questions you have I will do my best to help you.
